src/wp-includes/rest-api.php

```diff
@@ -1284,12 +1285,36 @@ function rest_validate_value_from_schema( $value, $args, $param = '' ) {
 			return new WP_Error( 'rest_invalid_param', sprintf( __( '%1$s is not of type %2$s.' ), $param, 'object' ) );
 		}
 
+		/*
+		 * @ticket 48818
+		 */
+		if ( isset( $args['properties'] ) && ! empty( $value ) ) {
+			if ( isset( $args['required'] ) ) { // schema version 4
+				foreach ( $args['required'] as $name ) {
+					if ( ! isset( $value[ $name ] ) ) {
+						/* translators: %s: Property of an object. */
+						return new WP_Error( 'rest_property_required', sprintf( __( '%1$s is a required property of Object.' ), $name ) );
+					}
+				}
+			} else { // schema version 3
+				foreach ( $args['properties'] as $name => $property ) {
+					if ( isset( $property['required'] ) && true === (bool) $property['required'] && ! isset( $value[ $name ] ) ) {
```

Comment:
```suggestion
					if ( isset( $property['required'] ) && true === $property['required'] && ! isset( $value[ $name ] ) ) {
```

Casting this to `bool` removes the purpose of the strict equality check and would treat the v4 format as the v3 format.

Reply:
Yes correct